doc: Add Applying task migrations (#439)
konflux-ci/docs: modules/building/pages/apply-task-migrations.adoc

= Applying task migrations

{ProductName} allows task owners to write migrations to bring pipeline changes
to users. Generally, users do not need to care about how the migrations are
defined and how {ProductName} applies migrations to their build pipelines.

> MintMaker applies the migrations automatically as part of the regular pull requests that update Tekton tasks. However, in case users requires to apply

Reviewer: ```suggestion
MintMaker applies the migrations automatically as part of the regular pull requests that update Tekton tasks. However, in case users are required to apply
```

Renovate update pull request together. However, in case users requires to apply
migrations by themselves, this document illustrates how to do that.

== Apply migration with pipeline-migraiton-tool

. Install pipeline-migration-tool:

+
Choose a release from
https://github.com/konflux-ci/pipeline-migration-tool/releases[releases] page
and install a specific version.

+
[source,bash]
----
pipx install https://github.com/konflux-ci/pipeline-migration-tool/archive/refs/tags/v0.4.2.zip
----

+

> then, there are two executables available in the `$PATH`,
> `pipeline-migration-tool` and `pmt`. Each of them is usable. In this document,
> `pmt` will be used.

Reviewer 2: Slight rewording to make it clear the `pipeline-migration-tool` and `pmt` are the same thing and to use active voice rather than passive voice

```suggestion
This installs two executables in `$PATH` - `pipeline-migration-tool` and its alias, `pmt`.
In this document, we will use `pmt`.
```

  upgrade data:

+
[source,bash]
----
cd path/to/component-repo

pmt migrate -u '
[
    {
        "depName": "quay.io/konflux-ci/tekton-catalog/task-clamav-scan",
        "currentValue": "0.2",
        "currentDigest": "sha256:7749146f7e4fe530846f1b15c9366178ec9f44776ef1922a60d3e7e2b8c6426b",
        "newValue": "0.3",
        "newDigest": "sha256:cce2dfcc5bd6e91ee54aacdadad523b013eeae5cdaa7f6a4624b8cbcc040f439",
        "packageFile": ".tekton/<pipeline file>", <.>
        "parentDir": ".tekton",
        "depTypes": ["tekton-bundle"]
    },
    {
        "depName": "quay.io/konflux-ci/tekton-catalog/task-clamav-scan",
        "currentValue": "0.2",
        "currentDigest": "sha256:7749146f7e4fe530846f1b15c9366178ec9f44776ef1922a60d3e7e2b8c6426b",
        "newValue": "0.3",
        "newDigest": "sha256:cce2dfcc5bd6e91ee54aacdadad523b013eeae5cdaa7f6a4624b8cbcc040f439",
        "packageFile": ".tekton/<pipeline file>", <.>
        "parentDir": ".tekton",
        "depTypes": ["tekton-bundle"]
    }
    , ... <.>
]
'
----

+
<.> Replace `<pipeline file>` with actual file name which is under `.tekton/` directory.
<.> Likewise. Specify another pipeline file.

> <.> More upgrades can be specified if there is.

Reviewer 2: ```suggestion
<.> You can specify more upgrades if there are any.
```

[NOTE]
====

> When saying build pipeline and pipeline file, it means a YAML file including
> tekton `Pipeline` definition or `PipelineRun` definition where `Pipeline` is
> embedded in `.spec.pipelineSpec`.

Reviewer: ```suggestion
In this context, 'build pipeline' or 'pipeline file' are refering to a YAML file including a
Tekton `Pipeline` or `PipelineRun` definition.
```

I feel like explaining what a pipelineRun is here is confusing. I would refer users to https://tekton.dev/docs/concepts/concept-model/#steps-tasks-and-pipelines as a last resort.

tekton `Pipeline` definition or `PipelineRun` definition where `Pipeline` is
embedded in `.spec.pipelineSpec`.
====

> . For whatever the reason, in case there is no migration delivered by the

Reviewer 2: Could we add a heading for this section?

```
== Apply migration manually
```

Author: Updated. PTAL.

>   bundle update. The linked `migration` document always includes instructions

Reviewer 2: ```suggestion
  bundle update. The linked `migration` document should include instructions
```

It would be nice if they always included instructions, but they're written by humans and often don't include useful information, unfortunately
